### ISSUE TITLE
type(range(10)) and type(zip([], [])) is incorrect in python3

### ISSUE THREAD

**Issue by ANDEMEN** [skulpt/skulpt#979]
in real python3:

```python
type(range(10))    # <class 'range'>
type(zip([], []))  # <class 'zip'>
```

in skulpt with python3 mode:

```python
type(range(10))    # <class 'list'>
type(zip([], []))  # <class 'list'>
```

do you have any plan to fix it ?

**Comment by Dustyposa** [comment]
not a easy function. Have not generator. 
But change the type data is the fast way to fix this problem. It's ok?

**Comment by acbart** [comment]
Actually, this is an intentional design decision so far motivated by the pedagogical nature of Skulpt. I actually even agree with it - having to explain generators early on or hand-wave them is a big pain. I suppose they could build a flag to change this behavior, since I'm pretty sure generators work fine.

**Comment by albertjan** [comment]
@acbart you give us more credit than we deserve 😆. It's this way because this is what happens in python2 and we haven't implemented the new versions yet. There might be a pedagogical reason to have list implementations too. But I also think that we shouldn't diverge too far from python it could be frustrating for students to find that things they wrote in skulpt doesn't work in python. 

I'm sure @meredydd would love to have these as proper generators so you can actually zip asynchronous streams.

**Comment by s-cork** [comment]
Closed with #938 and #1024
